www/js/metrics.js

```diff
@@ -699,77 +702,91 @@ angular.module('emission.main.metrics',['nvd3', 'emission.services', 'ionic-date
    $scope.fillFootprintCardUserVals = function(userDistance, twoWeeksAgoDistance) {
       if (userDistance) {
         var userCarbonData = getSummaryDataRaw(userDistance, 'distance');
+        console.debug('TIAGO: fillFootprintCardUserVals userDistance ' + JSON.stringify(userDistance, null, 2));
+        console.debug("TIAGO: fillFootprintCardUserVals userCarbonData (distances, from getSummaryDataRaw)" + JSON.stringify(userCarbonData, null, 2));
+
         var optimalDistance = getOptimalFootprintDistance(userDistance);
-        var worstDistance = getWorstFootprintDistance(userDistance);
+        var worstDistance   = getWorstFootprintDistance(userDistance);
         var date1 = $scope.selectCtrl.fromDateTimestamp;
         var date2 = $scope.selectCtrl.toDateTimestamp;
         var duration = moment.duration(date2.diff(date1));
         var days = duration.asDays();
-        //$scope.ca2020 = 43.771628 / 5 * days; // kg/day
+
         $scope.carbonData.ca2035 = Math.round(40.142892 / 5 * days) + ' kg CO₂'; // kg/day
         $scope.carbonData.ca2050 = Math.round(8.28565 / 5 * days) + ' kg CO₂';
-        //$scope.carbonData.userCarbon = [];
-        for (var i in userCarbonData) {
-          //$scope.carbonData.userCarbon.push({key: userCarbonData[i].key, values: FootprintHelper.getFootprint(userCarbonData[i].values, userCarbonData[i].key)});
-          if (userCarbonData[i].key === "IN_VEHICLE") {
-            $scope.carbonData.userCarbon = FootprintHelper.getFootprint(userCarbonData[i].values, userCarbonData[i].key);
-            $scope.carbonData.optimalCarbon = FootprintHelper.getFootprint(optimalDistance, userCarbonData[i].key);
-            $scope.carbonData.worstCarbon = FootprintHelper.getFootprint(worstDistance, userCarbonData[i].key);
-            lastWeekCarbonInt = FootprintHelper.getFootprintRaw(userCarbonData[i].values, userCarbonData[i].key);
-          }
-        }
+
+        $scope.carbonData.userCarbon    = FootprintHelper.readableFormat(FootprintHelper.getFootprintForMetrics(userCarbonData));
```

Comment:
Big difference here is that these are individual values and not arrays any more. So we can also make them readable right now.